### PR TITLE
JAX implicitly restarted arnoldi

### PR DESCRIPTION
**Pull request by mganahl**
fix convergence test, add tests

### CODE REVIEW

File: tensornetwork/backends/jax/jax_backend_test.py [hunk 1]
```diff
@@ -768,10 +795,121 @@ def mv(x, H):
...
+    t1 = neye * pot[N - 2] / 2 + eyen * pot[N - 1]
+    t2 = cTc * hop[N - 2] - ccT * jnp.conj(hop[N - 2])
+    out += jnp.einsum('ij,ki -> kj', x, t1 + t2)
+    x = x.reshape((2, 2**(N - 1))).transpose((1, 0)).reshape((4, 2**(N - 2)))
+    out = out.reshape((2, 2**(N - 1))).transpose((1, 0)).reshape(
+        (4, 2**(N - 2)))
+
+    x = x.reshape((2, 2**(N - 1))).transpose((1, 0)).reshape(2**N)
+    out = out.reshape((2, 2**(N - 1))).transpose((1, 0)).reshape(2**N)
+    return out.ravel()
```
Comment (Thenerdstation) [inline tensornetwork/backends/jax/jax_backend_test.py:881]: What in the world is happening here lol.

Comment (mganahl) [reply tensornetwork/backends/jax/jax_backend_test.py:881]: This is the matric vector product for the transverse field Ising Hamiltonian on open boundary conditions!

**Comment by mganahl** [comment]
@alewis @Thenerdstation ping for review!

File: tensornetwork/backends/jax/jax_backend_test.py (hunk 1 above)
Comment (Thenerdstation) [reply tensornetwork/backends/jax/jax_backend_test.py:881]: I'll take your word for it.

**Review by Thenerdstation — APPROVED** [review]

**Pull request merged**